mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -338,13 +345,37 @@ def _get_workspace_id(self, run_id):
                 "current user. No run link will be recorded for the model version."
             )
             return None
-        if _DATABRICKS_ORG_ID_HEADER not in response.headers:
+        return response
+
+    def _get_workspace_id(self, get_run_response_proto):
+        if (
+            get_run_response_proto is None
+            or _DATABRICKS_ORG_ID_HEADER not in get_run_response_proto.headers
+        ):
             _logger.warning(
                 "Unable to get model version source run's workspace ID from request headers. "
                 "No run link will be recorded for the model version"
             )
             return None
-        return response.headers[_DATABRICKS_ORG_ID_HEADER]
+        return get_run_response_proto.headers[_DATABRICKS_ORG_ID_HEADER]
+
+    def _get_notebook_id(self, get_run_response_proto):
+        if get_run_response_proto is None:
+            return None
+        notebook_id = None
+        try:
+            js_dict = json.loads(get_run_response_proto.text)
+            parsed_response = GetRun.Response()
+            parse_dict(js_dict=js_dict, message=parsed_response)
+            run = Run.from_proto(parsed_response.run)
```

Comment:
Could we move this parsing logic into `_get_run`? Also assuming the HTTP request to get the run succeeds (`verify_rest_response` succeeds so we have actually fetched the run), we should probably throw if we can't deserialize the run as JSON, that shouldn't ever happen